### PR TITLE
Fix block-inserted-events e2e testing

### PR DESCRIPTION
**Pull request by edanzer**
### Proposed Changes

This PR fixes broken tests for editor-tracking > block-inserted-events. 

Note: the last test is just being skipped. I've added a comment in the file explaining why, and would plan to return to that test based on what we decide to do. 

### Testing Instructions

1) Set up your testing environment following the instructions here: https://github.com/Automattic/wp-calypso/tree/trunk/test/e2e

2) One set up, run the relevant tests with `yarn jest specs/editor-tracking/editor-tracking__block-inserted-events`. You should see 1 of 1 test suites pass and 26 of 28 tests pass with 2 skipped.

### CODE REVIEW

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (edanzer) [inline test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: This property does not appear to exist on this event anymore. Not sure when that changed, but given that, I'm just removing the non-present property.

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (edanzer) [inline test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: Again, this property does not appear to exist on this event anymore. Not sure when that changed, but given that, I'm just removing the non-present property.

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (Addison-Stavlo) [reply test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: I took a look and it seems the entity_context is a quick fix. It seems to be a result of a element ID change in the editor. https://github.com/Automattic/wp-calypso/pull/69184 should resolve the issue.

**Comment by dpasque** [comment]
Hey @edanzer! 👋 

Following up from this thread: p1665714387215589-slack-C1A1EKDGQ

... the only change that you'll ultimately need is skipping the block_inserted check until we decide how that should behave, correct?

If so, that looks good to me! 😄

**Review by dpasque — APPROVED** [review]
Good to go once the `entity_context` pieces are taken out! 👍

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (edanzer) [inline test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: Upon further investigation, the getActiveBlockVariation() method is working correctly for generic template parts in a vanilla, self-hosted WordPress install. The issue noted here, where every template part inserted generates a new block variation, seems to be unique to WordPress.com. I would plan to add a follow up issue with details.

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (edanzer) [inline test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts:292]: I would plan to add a follow up issue on this, to be addressed separately once we decide what the expected behavior should be.

**Comment by dpasque** [comment]
@edanzer Updates look good! Oof, this seems like this has been hard to wrestle into submission with the bugs that have crept in, but I think you're leaving the tests in their best spot for now. 👍 

If you have issues spun up for the related bugs, I recommend adding those inline in the comments before you merge!

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (Addison-Stavlo) [reply test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: The bug is not unique to wordpress.com

[core wordpress is initializing every template part that exists as a block variation](https://github.com/WordPress/wordpress-develop/blob/755d5baff3fde70ebd2faf433a0e0cb46dd12a56/src/wp-includes/blocks/template-part.php#L191), and the [template part block has an outdated isActive function that still matches based on the area classification](https://github.com/WordPress/gutenberg/blob/5a82cf5723b0feb0de9ec23e267f6b190d2a5ad0/packages/block-library/src/template-part/variations.js#L30). I can reproduce this on my own self hosted docker instance running WP 6.0.3 and gutenberg `trunk`. You may just need to reload the editor after creating/saving a new template part, since block variations are not updated after initial block registration somewhere in the initialization process.

Its ok to skip this for now. but we can fix `entity_context` by updating our tracking event handling to append the `area` property of the template part instead of the variation slug as described in p1666206727128859/1666121794.413879-slack-CHN6J22MP

File: packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts
Comment (edanzer) [inline packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts:12]: This was broken recently due to a [change in Gutenberg](https://github.com/WordPress/gutenberg/commit/569241a3e1b7d0cc0567fccb92b1b57eb1b5b9c5#diff-0f64190407e5a202cfef71e90e6065e3c7f6ad4fcd8601e9f3f85df137b11a92R149). They updated the selector for the block inserter button for edit-site only. The selector was changed from `edit-site-header-toolbar__inserter-toggle` to `edit-site-header-edit-mode__inserter-toggle`. For edit-post, the selector is still `edit-post-header-toolbar__inserter-toggle`. 

The code changed here still needs to work correctly on both selectors. Given the differences, the text `inserter-toggle` is shared text. I was concerned that may be too generic, but it tests correctly for E2E tests in both the edit-site and edit-post environments.

**Comment by edanzer** [comment]
There have been a couple updates to this PR since prior rounds of testing.
- I recently merged a separate [PR to adjust how we set entity_context for template parts](https://github.com/Automattic/wp-calypso/pull/69398). This allowed us to reenable a test here we had been planning to skip. 
- Gutenberg [recently merged a change](https://github.com/WordPress/gutenberg/commit/569241a3e1b7d0cc0567fccb92b1b57eb1b5b9c5#diff-0f64190407e5a202cfef71e90e6065e3c7f6ad4fcd8601e9f3f85df137b11a92R149) to the css selector for the block inserter button on the page editor. That change started breaking some tests in this file. I've committed a fix and added a comment [here](https://github.com/Automattic/wp-calypso/pull/69154#discussion_r1010560045).

File: test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts
Comment (Addison-Stavlo) [inline test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: It looks like we are adding new tests here as well?

The first seems like it is just setup for the second? And it doesn't expect/assert anything. I wonder if it makes more sense to combine those into one test or actually `expect` something in that first one?

Comment (edanzer) [reply test/e2e/specs/editor-tracking/editor-tracking__block-inserted-event.ts]: We had added these tests at a point where we were going to be skipping other tests in this file that tested inserted blocks within general template parts. Those tests were not working due to issues with entity_context. Header template parts did not have the same issues, so I recreated the tests here. 

We've since resolved the underlying entity_context issue and reenabled skipped tests for general template parts. I had been thinking to keep these new tests as they're working and just add more coverage. But after discussion, we've decided it would be better to avoid adding redundant tests that we'll need to maintain going forward. 

So I'll remove these now. Thanks!

File: packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts
Comment (dpasque) [reply packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts:12]: This seems like a good change to me! I'm going to kick off a few more E2E test builds to just make sure there's not any unexpected failures there 👍

**Comment by dpasque** [comment]
@edanzer Final state LGTM! 🎉  Thank you for all your iterations on this! 😄

**Comment by edanzer** [comment]
Related issue: https://github.com/Automattic/wp-calypso/issues/69254

**Pull request merged**
